torchtune/training/checkpointing/_utils.py

```diff
@@ -277,3 +298,266 @@ def update_state_dict_for_classifier(
+        resume_from_checkpoint (bool): Whether to resume from a checkpoint.
+        has_adapter_checkpoint (bool): Indicates if there is an adapter checkpoint.
+    Returns:
+        list[Path]: Sorted list of paths to the checkpoint files.
+    Example:
+        >>> checkpoint_files = ["model-00001-of-00002.safetensors", "model-00002-of-00002.safetensors"]
+        >>> checkpoint_dir = "/path/to/checkpoints"
+        >>> output_dir = "/path/to/output"
+        >>> resume_from_checkpoint = True
+        >>> has_adapter_checkpoint = False
+        >>> paths = get_model_checkpoint_path(
+        ...     checkpoint_files,
+        ...     checkpoint_dir,
+        ...     output_dir,
+        ...     resume_from_checkpoint,
+        ...     has_adapter_checkpoint
+        ... )
+        >>> print(paths)
+        [PosixPath('/path/to/output/{largest_epoch}/model-00001-of-00002.safetensors'),
+         PosixPath('/path/to/output/{largest_epoch}/model-00002-of-00002.safetensors')]
+    """
+
+    def validate_checkpoint_files(
+        checkpoint_files: Union[List[str]],
+        input_dir: Optional[Path] = None,
+        missing_ok=False,
+    ) -> List[Path]:
+        """
+        Validates that the checkpoint files exist and sorts based on ID.
+        """
+        if not input_dir:
+            input_dir = self._checkpoint_dir
+
+        checkpoint_paths: List[Path] = []
+        for f in checkpoint_files:
+            checkpoint_path = get_path(input_dir, f, missing_ok)
+            checkpoint_paths.append(checkpoint_path)
+
+        return sorted(checkpoint_paths)
+
+    # load or resume from model weights
+
+    # e.g.
+    # checkpoint_files:
+    #   filename_format: model-{}-of-{}.safetensors
+    #   max_filename: 00191
+    # becomes checkpoint_files = [model-00001-of-00191.safetensors, model-00002-of-00191,..]
+    if not isinstance(checkpoint_files, List):
+        # TODO: this can be a function instead of a class
+        formatted_checkpoint_files = FormattedCheckpointFiles.from_dict(
+            checkpoint_files
+        )
+        checkpoint_files = formatted_checkpoint_files.build_checkpoint_filenames()
+
+    # Case 1: no resuming from ckpt
+    if not resume_from_checkpoint:
+        input_dir = checkpoint_dir
+
+    # Case 2: Resuming from ckpt, but its full finetuning (no adapter)
+    elif has_adapter_checkpoint is None:
+        input_dir = output_dir
+
+    # Case 3: Resuming from ckpt and has an adapter.
+    else:
+        # FIXME
+        # TODO: if the model has lora + trained weights, e.g. embeddings,
+        # we will silently not load the trained model, because we load from checkpoint_dir.
+        # We cannot load from output_dir because we always merge the adapter weights into the model
+        input_dir = checkpoint_dir
+
+    checkpoint_paths = validate_checkpoint_files(
+        checkpoint_files,
+        input_dir=checkpoint_dir,
```

Comment:
I'm confused - you set input_dir with all the above switch statements and then you don't use it here?

Reply:
-.-`

Reply:
i will update a couple of unit tests to use different output dir